### DISCUSSION TITLE
[General] How do I get L1 finalize block count

### DISCUSSION THREAD

**Discussion by blueberry6401** [ethereum-optimism/developers#341] [General]
### Discussion Topic

Hello,
What I want to do is quite like this: Binance will acknowledge the deposit when there is 64 block confirmations on L1.
<img width="662" alt="image" src="https://github.com/ethereum-optimism/developers/assets/7337535/0de940d0-6053-4ff0-82b1-f15bdd16ee2a">

So how do I check how many block is finalized on L1 from a L2 transaction?
Thank.

**Comment by imtipi** [comment]
![image](https://github.com/ethereum-optimism/developers/assets/27230181/ae5d63ec-81a0-45d4-8a95-758a9f55fb8f)
i never can't get the mean of bundle1/2..

> So how do I check how many block is finalized on L1 from a L2 transaction?

first get blocknumber from hash(etherscan/eth_getTransactionReceipt)
then compare the blocknumber between `eth_getBlockByNumber` with the parameter finalized as the block number.

resource:https://docs.optimism.io/builders/app-developers/transactions/statuses#finalized

> Finalized
> Typically within 15-20 minutes, up to 24 hours
> 
> A transaction is considered "finalized" when it has been included in a block by the Sequencer, that block has been published to Ethereum, and that block has been finalized. Once a block has been finalized it is guaranteed to be included in the OP Mainnet blockchain. Applications that require the highest level of certainty that a transaction will be included in the blockchain should wait until the transaction is "finalized" before considering the transaction to be successful.
> 
> The latest "finalized" block can be retrieved by calling the standard JSON-RPC method eth_getBlockByNumber with the parameter finalized as the block number.

**Comment by IvanKodak** [comment]
@imtipi your answer is correct, but it does not cover the case when you want to count L1 confirmations of L2 transaction.
For example, how can I get the L1 block number in which included my L2 transaction?

**Reply by joohhnnn** [reply]
> @imtipi your answer is correct, but it does not cover the case when you want to count L1 confirmations of L2 transaction. For example, how can I get the L1 block number in which included my L2 transaction?

You can filter out the L1 blocks that contain batch transactions from the batcher, and then use `optimism_safeHeadAtL1Block `to determine which L1 block corresponds to the L2 block. cc @imtipi @blueberry6401
